libs/api/src/main/java/de/cyface/api/ClassifiedSegmentRetriever.java

```diff
@@ -50,30 +53,61 @@ public class ClassifiedSegmentRetriever {
      * The name of the collection in the Mongo database used to store classified segments data.
      */
     private final String collectionName;
+    /**
+     * The query which defines which data to load.
+     */
+    private final JsonObject query;
+    /**
+     * The aggregation mode.
+     */
+    private final ClassifiedSegmentFactoryProvider.Mode mode;
+
+    /**
+     * Constructs a fully initialized object of this object.
+     *
+     * @param collectionName The name of the collection in the Mongo database used to store deserialized measurements.
+     * @param mode The aggregation mode.
+     * @param segmentIds The identifiers of the segments to loaded
+     */
+    public ClassifiedSegmentRetriever(final String collectionName, final ClassifiedSegmentFactoryProvider.Mode mode,
```

Comment:
I need to inject the `mode` (measurement-based or user-based) to choose a Factory below.